primitives/plutus-data/src/version_oracle.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{DataDecodingError, DecodingResult, PlutusDataExtensions};
+use cardano_serialization_lib::{BigInt, PlutusData, PlutusList};
+
+/// Datum attached to 'VersionOraclePolicy' tokens stored on the 'VersionOracleValidator' script.
+/// This datum is not versioned intentionally, as it is not subject to observation.
+///
+/// Original definition in the smart contracts:
+/// ```haskell
+/// data VersionOracleDatum = VersionOracleDatum
+/// { versionOracle :: VersionOracle
+/// -- ^ VersionOracle which identifies the script.
+/// , currencySymbol :: CurrencySymbol
+/// -- ^ Currency Symbol of the VersioningOraclePolicy tokens.
+/// }
+///
+/// See https://preview.cexplorer.io/tx/70923772056f153d646488c56ac04d1bc2f1326f074773e4f262c63e03b72a3d/contract#data
+/// for an example of transaction outputing this datum.
+/// ```
+#[derive(Clone, Debug, PartialEq)]
+pub struct VersionOracleDatum {
+	pub version_oracle: u32,
+	pub currency_symbol: Vec<u8>,
```

Comment:
Is currency symbol arbitrary length or 28 bytes?

Reply:
It is arbitrary lenght in the haskell code.

Reply:
In offchain code it ScriptHash from CardanoSerializationLibrary (first you go to CTL, where it is defined as ScriptHash from purescript-cardano-types, where it is defined as ScriptHash from CSL). So, it is 28 bytes.

In haskell it is, by docs 28 bytes or empty for Ada. I think we should use 28 bytes here